### PR TITLE
unaligned loads and stores for LJFunctorAVX

### LINKED ISSUE #535: lc_combined* Traversals segfault with AVX Functor
**Describe the bug**

When using the lc_combined (`lc_c04_combined_SoA` and `lc_c01_combined_SoA` traversals in combination with the intrinsics functor sometimes segfaults happen in memory load operations.

**To Reproduce**
`md-flexible --no-end-config --no-flops --functor lennard-jones-AVX2 --deltaT 0 --particle-generator uniform --log-level debug --traversal lc_c04_combined_SoA  --particles-total 71`

**Expected behavior**
No segfault 🙃 

~**Screenshots**~

**Build environment (please complete the following information):**
* OS: WSL, CooLMUC2
* Compiler: (e.g. gcc 9, clang 11)
* Build System: make

**Additional context**
This might be due to these traversals not using SoAs directly but views. These views work as offsets from normal views which might break the alignment...?

### PR DESCRIPTION
**Pull request by SteffenSeckler**
# Description

Replaces unaligned loads and stores in LJFunctorAVX to be able to simulate with non-aligned SoAs from SoAViews (combined SoAs)

## Todo

Should probably be implemented using a template parameter to decide whether aligned or unaligned intrinsics should be used.

## Resolved Issues

- fixes md-flexible not being build if mpi is not present.
- [x] fixes #535 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] Ran example:
```bash
   ./md-flexible --no-end-config --no-flops --functor lennard-jones-AVX2 --deltaT 0 --particle-generator uniform --log-level debug --traversal lc_c04_combined_SoA --particles-total 71
```
- [x] Added the above example to the ctest tests

### CODE REVIEW

**Comment by FG-TUM** [comment]
What is the performance impact of this? According to Intel Intrinsics Guide both `load` and `loadU` are the same speed but I guess since `loadU` has to potentially fetch more than one cache line it might be slower if that is the case?

**Comment by SteffenSeckler** [comment]
lets use our performance testing for this :P

**Comment by FG-TUM** [comment]
IIRC it uses the regular LJFunctor so you might want to check things manually or talk to @kruegener if we can have some options here ;)

**Comment by kruegener** [comment]
I mean the perf measurements without selecting a specific functor ran fine ;)
I'll add an option to specify any additional CLI options for the upcoming web interface in october.

Do the CLI parameters overwrite yaml settings if both define e.g. a functor?

**Comment by FG-TUM** [comment]
yes, md-flex was intentionally designed so that CLI parameters always overwrite yaml parameters.

**Comment by kruegener** [comment]
> yes, md-flex was intentionally designed so that CLI parameters always overwrite yaml parameters.

Perfect. Will be easy then

**Comment by SteffenSeckler** [comment]
I've added a test to the examples (ctest). This test fails for the master and runs successfully on this PR.

**Comment by SteffenSeckler** [comment]
without testing too deep: performance looks equal.

**Comment by SteffenSeckler** [comment]
according to https://community.intel.com/t5/Intel-ISA-Extensions/SSE-and-AVX-behavior-with-aligned-unaligned-instructions/m-p/1170006/highlight/true#M6647 it appears that since haswell, large performance impacts are not to be expected.

**Comment by SteffenSeckler** [comment]
@FG-TUM ideally I would like a unit test for this :)

**Comment by FG-TUM** [comment]
@SteffenSeckler feel free to write one :P

**Comment by SteffenSeckler** [comment]
> @SteffenSeckler feel free to write one :P

done

File: examples/md-flexible/CMakeLists.txt [hunk 1]
```diff
@@ -92,10 +95,55 @@ add_test(
         CONFIGURATIONS checkExamples
 )
 
+set(HAS_AVX true)
```
Comment (FG-TUM) [inline examples/md-flexible/CMakeLists.txt:98]: Write a short justification comment for this whole code block. Would be nice to know what is coming without reading 50 lines

**Review by FG-TUM — APPROVED** [review]
just a bit more doc

File: examples/md-flexible/CMakeLists.txt [hunk 2]
```diff
@@ -95,27 +95,28 @@ add_test(
...
-        # set to false, as we don't know...
-        set(HAS_AVX false)
-    endif()
-else()
+    try_compile(
+            HAS_AVX
+            ${CMAKE_BINARY_DIR}
+            ${AUTOPAS_SOURCE_DIR}/cmake/tests/has_avx_test.cpp
+            # We are abusing COMPILE_DEFINITIONS here. There is no other sane way of passing "-march=native" ...
+            COMPILE_DEFINITIONS "-march=native"
```
Comment (SteffenSeckler) [inline examples/md-flexible/CMakeLists.txt:116]: found similar thing on `apple.com` so passing this using COMPILE_DEFINITIONS seems right...

https://opensource.apple.com/source/clang/clang-700.1.81/src/projects/compiler-rt/cmake/config-ix.cmake.auto.html

**Pull request merged**